stellargraph/core/element_data.py

```diff
@@ -331,28 +358,41 @@ def __init__(self, shared, type_starts):
 
     def _init_directed_adj_lists(self):
         # record the edge ilocs of incoming, outgoing and both-direction edges
-        in_dict = {}
-        out_dict = {}
+        in_dict = dict((node, []) for node in np.sort(pd.unique(self.targets)))
+        out_dict = dict((node, []) for node in np.sort(pd.unique(self.sources)))
+
+        dtype = np.min_scalar_type(len(self.sources))
 
         for i, (src, tgt) in enumerate(zip(self.sources, self.targets)):
-            in_dict.setdefault(tgt, []).append(i)
-            out_dict.setdefault(src, []).append(i)
+            in_dict[tgt].append(i)
+            out_dict[src].append(i)
 
-        dtype = np.min_scalar_type(len(self.sources))
-        self._edges_in_dict = _numpyise(in_dict, dtype=dtype)
-        self._edges_out_dict = _numpyise(out_dict, dtype=dtype)
+        in_neighbour_counts = np.array([len(x) for x in in_dict.values()])
```

Comment:
(This `np.array` code seems like it could be inside `FlatAdjacencyList`, since it's repeated and always just on the `adj_dict` argument.)